### PR TITLE
Refactor permissions functions

### PR DESCRIPTION
**Pull request by mkocher**
Now that space supporter implementation is complete we can collapse a
bunch of functions in permissions.rb and finalize naming conventions
that we're mostly happy with.

* include space supporter in ALL_PERMISSIONS
* removed space supporter specific mentions in permissions.rb

Co-authored-by: Mona Mohebbi <mmohebbi@pivotal.io>
Co-authored-by: Merric de Launey <mdelauney@pivotal.io>
Co-authored-by: Matthew Kocher <mkocher@pivotal.io>

Thanks for contributing to cloud_controller_ng. To speed up the process of reviewing your pull request please provide us with:

* A short explanation of the proposed change:

* An explanation of the use cases your change solves

* Links to any other associated PRs

* [X] I have reviewed the [contributing guide](https://github.com/cloudfoundry/cloud_controller_ng/blob/main/CONTRIBUTING.md)

* [X] I have viewed, signed, and submitted the Contributor License Agreement

* [X] I have made this pull request to the `main` branch

* [X] I have run all the unit tests using `bundle exec rake`

* [ ] I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cloud_controller_ng#cf-acceptance-tests-cats)

### CODE REVIEW

**Comment by sweinstein22** [comment]
I'm going to be OOO tomorrow but I'll plan to have a look at this on Wednesday. In the meantime just want to call out it looks like we've got some merge conflicts to resolve

**Comment by mkocher** [comment]
Thank you. We resolved the merge conflicts. /cc @MerricdeLauney

**Comment by mkocher** [comment]
Closes #2467

File: lib/cloud_controller/permissions.rb [hunk 1]
```diff
@@ -288,15 +262,15 @@ def can_read_system_environment_variables?(space_guid, org_guid)
       membership.has_any_roles?(ROLES_FOR_SPACE_SECRETS_READING, space_guid, org_guid)
   end
 
-  def untrusted_can_read_route?(space_guid, org_guid)
-    return true if can_read_globally?
+  # def can_read_route?(space_guid, org_guid)
```
Comment (sweinstein22) [inline lib/cloud_controller/permissions.rb:265]: Is there a reason we've left this method commented out rather than removing it?

**Review by sweinstein22 — CHANGES_REQUESTED** [review]
### Code Review

I'm noticing that for the most part we were just removing `supporter` or `untrusted` from method names, and then making the adjustment to `ALL_PERMISSIONS` to include `space_supporter`. The exceptions seem to be:
```
untrusted_can_write_to_space => can_manage_apps_in_space
untrusted_can_read_from_space => can_download_droplet
```
Looks like we updated the tests as appropriate for those two.


### Changes Requested
I did a quick search through the branch for `untrusted` and noticed we still seem to have mention of some of these methods in:
```
service_route_bindings_controller.rb
service_instances_controller.rb
```
I think addressing those would be part of the associated issue for this PR. I also left a comment on one section, looks like we left in a commented out method, just wondering if we left it for a specific reason.

File: lib/cloud_controller/permissions.rb (hunk 1 above)
Comment (monamohebbi) [reply lib/cloud_controller/permissions.rb:265]: No, good catch!

**Comment by monamohebbi** [comment]
this branch has gotten a little too messy. Opening a new pr with the right commits cherry picked on top

**Pull request closed without merging**